src/overlays/actors/ovl_En_Dnq/z_en_dnq.c

```diff
@@ -22,55 +52,424 @@ const ActorInit En_Dnq_InitVars = {
+    s32 ret = false;
+
+    if ((arg1 == 0) || (arg1 == 1)) {
+        switch (this->unk_398) {
+            case 0:
+            case 1:
+                break;
+
+            default:
+                phi_v1 = true;
+                break;
+        }
+    } else if (arg1 != this->unk_398) {
+        phi_v1 = true;
+    }
+
+    if (phi_v1) {
+        this->unk_398 = arg1;
+        ret = func_8013BC6C(&this->skelAnime, D_80A5349C, arg1);
+    }
+
+    return ret;
+}
+
+void func_80A52604(EnDnq* this, GlobalContext* globalCtx) {
+    Collider_UpdateCylinder(&this->actor, &this->collider);
+    CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+}
+
+s32 func_80A52648(EnDnq* this, GlobalContext* globalCtx) {
+    s32 ret = false;
+
+    if (globalCtx->csCtx.state != 0) {
+        if (!(this->unk_37C & 0x20)) {
+            this->actor.flags &= ~1;
+            this->unk_1DC = 0xFF;
+            this->unk_37C |= 0x20;
+        }
+        func_8013AED4(&this->unk_37C, 0, 7);
+        ret = true;
+    } else {
+        if (this->unk_37C & 0x20) {
+            this->actor.flags |= 1;
+            this->unk_1DC = 0xFF;
+            this->unk_37C &= ~0x20;
+            func_8013AED4(&this->unk_37C, 3, 7);
+        }
+    }
+
+    return ret;
+}
+
+s32 func_80A526F8(EnDnq* this) {
+    if (this->unk_386 == 0) {
+        func_80A5257C(this, 8);
+        this->unk_38C = ((s32)(Rand_ZeroOne() * 100.0f) % 4) + 3;
+        this->unk_388 = 0;
+        this->unk_386 = 1;
+    } else if (func_801378B8(&this->skelAnime, this->skelAnime.animFrameCount)) {
+        switch (this->unk_398) {
+            case 8:
+                if (DECR(this->unk_38C) == 0) {
+                    func_80A5257C(this, 9);
+                }
+                break;
+
+            case 9:
+                func_80A5257C(this, 10);
+                this->unk_38C = ((s32)(Rand_ZeroOne() * 100.0f) % 3) + 2;
+                break;
+
+            case 10:
+                if (DECR(this->unk_38C) == 0) {
+                    this->unk_388++;
+                    if ((this->unk_38C != 0) || (this->unk_388 < 2)) {
+                        func_80A5257C(this, 11);
+                    } else {
+                        func_80A5257C(this, 17);
+                        this->unk_38C = 4;
+                    }
+                }
+                break;
+
+            case 11:
+                func_80A5257C(this, 8);
+                this->unk_38C = ((s32)(Rand_ZeroOne() * 100.0f) % 4) + 3;
+                break;
+
+            case 17:
+                if (DECR(this->unk_38C) == 0) {
+                    this->unk_386 = 0;
+                }
+                break;
+        }
+
+        if (this->unk_386 != 0) {
+            this->skelAnime.animCurrentFrame = 0.0f;
+        }
+    }
+
+    this->unk_38A = 0;
+    return 0;
+}
+
+s32 func_80A52944(EnDnq* this) {
+    s16 phi_v1 = 0;
+
+    if (this->unk_386 == 0) {
+        func_80A5257C(this, 0xC);
```

Comment:
```suggestion
        func_80A5257C(this, 12);
```
Other uses of this function seem to be in dec, just for consistency